src/server.rs

```diff
@@ -76,39 +150,60 @@ pub struct Server<L, M>
 
     /// Currently registered reconnection timeouts.
     reconnection_timeouts: HashMap<Token, TimeoutHandle>,
+
+    /// Configured timeouts
+    timeout_config: TimeoutConfiguration,
 }
 
 /// The implementation of the Server.
 impl<L, M> Server<L, M>
     where L: Log,
           M: StateMachine
 {
+    pub fn new(
```

Comment:
Yes, but I think the naming could use some work.  I renamed `builder()` to `new()`, and the old `new()` to `create()`.  Note that `create()` is a non-public function.  It's just there to finalize the creation.  Maybe since `Server::new()` delegates to `ServerBuilder::new()`, I should make `ServerBuilder::finalize()` delegate to `Server::finalize()`, instead of `Server::create()`.  For the moment, that's all internal code, but it may become public if the run phase gets extracted, so it probably makes sense to give some thought to naming.